src/bmi.jl

```diff
@@ -65,10 +65,20 @@ end
 
 function BMI.update_until(model::Model, time::Float64)
     @unpack clock, network, config = model
-    curtime = BMI.get_current_time(model)
-    n = Int(max(0, (time - curtime) / model.clock.Δt.value))
-    for _ = 1:n
-        model = run_timestep(model)
+    t = BMI.get_current_time(model)
+    steps = (time - t) / model.clock.Δt.value
+    if steps < 0
+        error("The current model timestamp $t is larger than provided `time` $time")
+    elseif !isinteger(steps)
+        error_message = string(
+            "Provided `time` $time minus the current model timestamp $t",
+            " is not an integer multiple of model time step $(model.clock.Δt.value)",
+        )
+        error(error_message)
```

Comment:
I definitely agree that is a good idea to throw an error for the exact reason as stated in the error message.
However, to check whether `steps` is an integer indicates something is wrong / ill-defined somewhere else in the code.

I suggest to use `divrem` for computing `steps`.
You can use the value of the remainder for deciding whether to throw the error message.

    _div, _rem = divrem(time - curtime, model.clock.Δt.value)
    steps = Int(_div)

    if steps < 0
       # Throw error 
    elseif _rem > eps()
        # Throw error
    end

Reply:
Indeed `isinteger` does not help with floating point issues like mentioned in https://github.com/Deltares/Wflow.jl/pull/387#issuecomment-2039814894. There I suggested `isapprox(round(t), t)`, which I think is a bit better since it supports small deviations on either side where `divrem` only supports overshoot.

Reply:
`divrem` does support both undershoot and overshoot. Depending on the inputs the outputs can be positive or negative.

Reply:
Just discussed with @CFBaptista that `divrem` is also good (slightly better), as long as the undershoot is also handled with negative `div`.

Reply:
Though even when the internal time format is the same, Float64, I think it is very common to calculate that time using floating point arithmetic. Therefore I think it's better that we handle this ourselves, it is only a few lines of code extra.

Reply:
Ok, sounds good, went with `divrem` approach, see commit https://github.com/Deltares/Wflow.jl/pull/387/commits/da8948cd5dca574e049d3b43b1feace77dda4a16.